staging/src/k8s.io/kubectl/pkg/cmd/cmd.go

```diff
@@ -280,7 +280,10 @@ func NewKubectlCommand(in io.Reader, out, err io.Writer) *cobra.Command {
 
 	flags.BoolVar(&warningsAsErrors, "warnings-as-errors", warningsAsErrors, "Treat warnings received from the server as errors and exit with a non-zero exit code")
 
-	kubeConfigFlags := genericclioptions.NewConfigFlags(true).WithDeprecatedPasswordFlag()
+	// Disable discovery burst as per discussion in https://github.com/kubernetes/kubernetes/pull/105520.
+	// We'll let APF (https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1040-priority-and-fairness)
+	// handle it for us. Further improvements in discovery will be handled separately in https://github.com/kubernetes/kubectl/issues/1126.
+	kubeConfigFlags := genericclioptions.NewConfigFlags(true).WithDeprecatedPasswordFlag().WithDiscoveryBurst(-1)
```

Comment:
I don't think this will work - see my previous comment. When I run kubectl with this PR I get an error. I couldn't immediately find any evidence that -1 has special meaning to the rate limiter and suspect it's treating it as an actual burst value. This is why I proposed https://github.com/kubernetes/kubernetes/pull/106016 as an alternative approach.